examples/ui/toolbar.rs

```diff
@@ -0,0 +1,279 @@
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+struct ToolbarState {
+    selected: ButtonTags,
+}
+
+impl Default for ToolbarState {
+    fn default() -> Self {
+        Self {
+            selected: ButtonTags::Stop,
+        }
+    }
+}
+
+struct ToolbarEvent {
+    tag: ButtonTags,
+    message: Option<String>,
+    interaction: Interaction,
+}
+
+#[derive(Component, Clone, Hash, Debug, Eq, PartialEq)]
+enum ButtonTags {
+    Play,
+    Pause,
+    Stop,
+}
+
+#[derive(Component)]
+struct ToolbarMessage;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        // Only run the app when there is user input. This will significantly reduce CPU/GPU use.
+        .insert_resource(WinitSettings::desktop_app())
+        .add_startup_system(setup)
+        .add_state(ToolbarState::default())
+        .add_event::<ToolbarEvent>()
+        .add_system(button_interaction_system)
+        .add_system(toolbar_state_system)
+        .add_system(toolbar_message_system)
+        .add_system(toolbar_style_system)
+        .run();
+}
+
+const BG_COLOR: Color = Color::rgb(0.1, 0.1, 0.1);
+const NORMAL_BUTTON: Color = Color::rgb(0.15, 0.15, 0.15);
+const HOVERED_BUTTON: Color = Color::rgb(0.25, 0.25, 0.25);
+const PRESSED_BUTTON: Color = Color::rgb(0.4, 0.4, 0.8);
+const SELECTED_BUTTON: Color = Color::rgb(0.35, 0.35, 0.75);
+
+/// System: Manages the click interaction and styling of all the button elements
+///
+/// Help wanted: I had to remove `Changed<Interaction>` from the query so that
+/// the buttons that weren't clicked would update.
+///
+/// Is there a way of listening to changes in `ResMut<State<ToolbarState>>`?
+fn button_interaction_system(
+    mut interaction_query: Query<(&Interaction, &ButtonTags), (Changed<Interaction>, With<Button>)>,
+    mut ev_toolbar_button: EventWriter<ToolbarEvent>,
+) {
+    for (interaction, button_tag) in &mut interaction_query {
+        match *interaction {
+            Interaction::Clicked => {
+                let message = match button_tag {
+                    ButtonTags::Stop => None,
+                    ButtonTags::Play => Some("Playing...".to_string()),
+                    ButtonTags::Pause => Some("Pause...".to_string()),
+                };
+
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message,
+                    interaction: interaction.clone(),
+                });
+            }
+            Interaction::Hovered => {
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message: None,
+                    interaction: interaction.clone(),
+                });
+            }
+            Interaction::None => {
+                ev_toolbar_button.send(ToolbarEvent {
+                    tag: button_tag.clone(),
+                    message: None,
+                    interaction: interaction.clone(),
+                });
+            }
```

Comment:
Nit: I think clippy will complain about this not being 
```suggestion
            Interaction::Hovered | Interaction::None => {
                ev_toolbar_button.send(ToolbarEvent {
                    tag: button_tag.clone(),
                    message: None,
                    interaction: interaction.clone(),
                });
            }
```

Reply:
Ahh I didn't know this was possible, thanks!